wagtail/core/blocks/base.py

```diff
@@ -73,8 +73,21 @@ def all_blocks(self):
 
     def all_media(self):
         media = forms.Media()
+
+        # In cases where the same block definition appears multiple times within different
+        # container blocks (e.g. a RichTextBlock appearing at the top level of a StreamField
+        # as well as both sides of a StructBlock for producing two-column layouts), we will
+        # encounter identical media declarations. Adding these to the final combined media
+        # declaration would be redundant add processing time when determining the final media
+        # ordering. To avoid this, we keep a cache of previously-seen declarations and only
+        # add unique ones.
+        media_cache = set()
+
         for block in self.all_blocks():
-            media += block.media
+            key = block.media.__repr__()
```

Comment:
Need to check that `repr` is unique - e.g. it doesn't get cut off after 10 items in the list

Reply:
What about generating a hash like we do for renditions? https://github.com/wagtail/wagtail/blob/d48d3dfc5b14ef6fd66a1aa5ce652f11f27fb3a1/wagtail/images/models.py#L460

Reply:
I tested with a hash. Unfortunatly it's really too time consuming. As far I had tested (about 30 000 media entries, reduced to about 30 after the fix), __repr__() seems to be the best choice as signature of the media.